// @aws-sdk/client-lambda-browser/model/shapes/_FunctionConfiguration.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var _VpcConfigResponse_1 = require("./_VpcConfigResponse");
var _DeadLetterConfig_1 = require("./_DeadLetterConfig");
var _EnvironmentResponse_1 = require("./_EnvironmentResponse");
var _TracingConfigResponse_1 = require("./_TracingConfigResponse");
var _LayersReferenceList_1 = require("./_LayersReferenceList");
exports._FunctionConfiguration = {
    type: "structure",
    required: [],
    members: {
        FunctionName: {
            shape: {
                type: "string",
                min: 1
            }
        },
        FunctionArn: {
            shape: {
                type: "string"
            }
        },
        Runtime: {
            shape: {
                type: "string"
            }
        },
        Role: {
            shape: {
                type: "string"
            }
        },
        Handler: {
            shape: {
                type: "string"
            }
        },
        CodeSize: {
            shape: {
                type: "integer"
            }
        },
        Description: {
            shape: {
                type: "string"
            }
        },
        Timeout: {
            shape: {
                type: "integer",
                min: 1
            }
        },
        MemorySize: {
            shape: {
                type: "integer",
                min: 128
            }
        },
        LastModified: {
            shape: {
                type: "string"
            }
        },
        CodeSha256: {
            shape: {
                type: "string"
            }
        },
        Version: {
            shape: {
                type: "string",
                min: 1
            }
        },
        VpcConfig: {
            shape: _VpcConfigResponse_1._VpcConfigResponse
        },
        DeadLetterConfig: {
            shape: _DeadLetterConfig_1._DeadLetterConfig
        },
        Environment: {
            shape: _EnvironmentResponse_1._EnvironmentResponse
        },
        KMSKeyArn: {
            shape: {
                type: "string"
            }
        },
        TracingConfig: {
            shape: _TracingConfigResponse_1._TracingConfigResponse
        },
        MasterArn: {
            shape: {
                type: "string"
            }
        },
        RevisionId: {
            shape: {
                type: "string"
            }
        },
        Layers: {
            shape: _LayersReferenceList_1._LayersReferenceList
        }
    }
};
