// @aws-sdk/client-lambda-browser/model/shapes/DeleteEventSourceMappingOutput.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteEventSourceMappingOutput = {
    type: "structure",
    required: [],
    members: {
        UUID: {
            shape: {
                type: "string"
            }
        },
        BatchSize: {
            shape: {
                type: "integer",
                min: 1
            }
        },
        EventSourceArn: {
            shape: {
                type: "string"
            }
        },
        FunctionArn: {
            shape: {
                type: "string"
            }
        },
        LastModified: {
            shape: {
                type: "timestamp"
            }
        },
        LastProcessingResult: {
            shape: {
                type: "string"
            }
        },
        State: {
            shape: {
                type: "string"
            }
        },
        StateTransitionReason: {
            shape: {
                type: "string"
            }
        }
    }
};
