// @aws-sdk/client-lambda-browser/model/shapes/UpdateFunctionConfigurationInput.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var _VpcConfig_1 = require("./_VpcConfig");
var _Environment_1 = require("./_Environment");
var _DeadLetterConfig_1 = require("./_DeadLetterConfig");
var _TracingConfig_1 = require("./_TracingConfig");
var _LayerList_1 = require("./_LayerList");
exports.UpdateFunctionConfigurationInput = {
    type: "structure",
    required: ["FunctionName"],
    members: {
        FunctionName: {
            shape: {
                type: "string",
                min: 1
            },
            location: "uri",
            locationName: "FunctionName"
        },
        Role: {
            shape: {
                type: "string"
            }
        },
        Handler: {
            shape: {
                type: "string"
            }
        },
        Description: {
            shape: {
                type: "string"
            }
        },
        Timeout: {
            shape: {
                type: "integer",
                min: 1
            }
        },
        MemorySize: {
            shape: {
                type: "integer",
                min: 128
            }
        },
        VpcConfig: {
            shape: _VpcConfig_1._VpcConfig
        },
        Environment: {
            shape: _Environment_1._Environment
        },
        Runtime: {
            shape: {
                type: "string"
            }
        },
        DeadLetterConfig: {
            shape: _DeadLetterConfig_1._DeadLetterConfig
        },
        KMSKeyArn: {
            shape: {
                type: "string"
            }
        },
        TracingConfig: {
            shape: _TracingConfig_1._TracingConfig
        },
        RevisionId: {
            shape: {
                type: "string"
            }
        },
        Layers: {
            shape: _LayerList_1._LayerList
        }
    }
};
