// @aws-sdk/client-lambda-browser/model/operations/CreateEventSourceMapping.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var CreateEventSourceMappingInput_1 = require("../shapes/CreateEventSourceMappingInput");
var CreateEventSourceMappingOutput_1 = require("../shapes/CreateEventSourceMappingOutput");
var ServiceException_1 = require("../shapes/ServiceException");
var InvalidParameterValueException_1 = require("../shapes/InvalidParameterValueException");
var ResourceConflictException_1 = require("../shapes/ResourceConflictException");
var TooManyRequestsException_1 = require("../shapes/TooManyRequestsException");
var ResourceNotFoundException_1 = require("../shapes/ResourceNotFoundException");
var ServiceMetadata_1 = require("../ServiceMetadata");
exports.CreateEventSourceMapping = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: "CreateEventSourceMapping",
    http: {
        method: "POST",
        requestUri: "/2015-03-31/event-source-mappings/"
    },
    input: {
        shape: CreateEventSourceMappingInput_1.CreateEventSourceMappingInput
    },
    output: {
        shape: CreateEventSourceMappingOutput_1.CreateEventSourceMappingOutput
    },
    errors: [
        {
            shape: ServiceException_1.ServiceException
        },
        {
            shape: InvalidParameterValueException_1.InvalidParameterValueException
        },
        {
            shape: ResourceConflictException_1.ResourceConflictException
        },
        {
            shape: TooManyRequestsException_1.TooManyRequestsException
        },
        {
            shape: ResourceNotFoundException_1.ResourceNotFoundException
        }
    ]
};
