// @aws-sdk/client-lambda-browser/model/operations/DeleteFunction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var DeleteFunctionInput_1 = require("../shapes/DeleteFunctionInput");
var DeleteFunctionOutput_1 = require("../shapes/DeleteFunctionOutput");
var ServiceException_1 = require("../shapes/ServiceException");
var ResourceNotFoundException_1 = require("../shapes/ResourceNotFoundException");
var TooManyRequestsException_1 = require("../shapes/TooManyRequestsException");
var InvalidParameterValueException_1 = require("../shapes/InvalidParameterValueException");
var ResourceConflictException_1 = require("../shapes/ResourceConflictException");
var ServiceMetadata_1 = require("../ServiceMetadata");
exports.DeleteFunction = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: "DeleteFunction",
    http: {
        method: "DELETE",
        requestUri: "/2015-03-31/functions/{FunctionName}"
    },
    input: {
        shape: DeleteFunctionInput_1.DeleteFunctionInput
    },
    output: {
        shape: DeleteFunctionOutput_1.DeleteFunctionOutput
    },
    errors: [
        {
            shape: ServiceException_1.ServiceException
        },
        {
            shape: ResourceNotFoundException_1.ResourceNotFoundException
        },
        {
            shape: TooManyRequestsException_1.TooManyRequestsException
        },
        {
            shape: InvalidParameterValueException_1.InvalidParameterValueException
        },
        {
            shape: ResourceConflictException_1.ResourceConflictException
        }
    ]
};
