// @aws-sdk/client-lambda-browser/model/operations/DeleteEventSourceMapping.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var DeleteEventSourceMappingInput_1 = require("../shapes/DeleteEventSourceMappingInput");
var DeleteEventSourceMappingOutput_1 = require("../shapes/DeleteEventSourceMappingOutput");
var ServiceException_1 = require("../shapes/ServiceException");
var ResourceNotFoundException_1 = require("../shapes/ResourceNotFoundException");
var InvalidParameterValueException_1 = require("../shapes/InvalidParameterValueException");
var TooManyRequestsException_1 = require("../shapes/TooManyRequestsException");
var ResourceInUseException_1 = require("../shapes/ResourceInUseException");
var ServiceMetadata_1 = require("../ServiceMetadata");
exports.DeleteEventSourceMapping = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: "DeleteEventSourceMapping",
    http: {
        method: "DELETE",
        requestUri: "/2015-03-31/event-source-mappings/{UUID}"
    },
    input: {
        shape: DeleteEventSourceMappingInput_1.DeleteEventSourceMappingInput
    },
    output: {
        shape: DeleteEventSourceMappingOutput_1.DeleteEventSourceMappingOutput
    },
    errors: [
        {
            shape: ServiceException_1.ServiceException
        },
        {
            shape: ResourceNotFoundException_1.ResourceNotFoundException
        },
        {
            shape: InvalidParameterValueException_1.InvalidParameterValueException
        },
        {
            shape: TooManyRequestsException_1.TooManyRequestsException
        },
        {
            shape: ResourceInUseException_1.ResourceInUseException
        }
    ]
};
