// @aws-sdk/client-lambda-browser/model/operations/DeleteAlias.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var DeleteAliasInput_1 = require("../shapes/DeleteAliasInput");
var DeleteAliasOutput_1 = require("../shapes/DeleteAliasOutput");
var ServiceException_1 = require("../shapes/ServiceException");
var InvalidParameterValueException_1 = require("../shapes/InvalidParameterValueException");
var TooManyRequestsException_1 = require("../shapes/TooManyRequestsException");
var ServiceMetadata_1 = require("../ServiceMetadata");
exports.DeleteAlias = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: "DeleteAlias",
    http: {
        method: "DELETE",
        requestUri: "/2015-03-31/functions/{FunctionName}/aliases/{Name}"
    },
    input: {
        shape: DeleteAliasInput_1.DeleteAliasInput
    },
    output: {
        shape: DeleteAliasOutput_1.DeleteAliasOutput
    },
    errors: [
        {
            shape: ServiceException_1.ServiceException
        },
        {
            shape: InvalidParameterValueException_1.InvalidParameterValueException
        },
        {
            shape: TooManyRequestsException_1.TooManyRequestsException
        }
    ]
};
