// @aws-sdk/client-lambda-browser/model/operations/GetAlias.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var GetAliasInput_1 = require("../shapes/GetAliasInput");
var GetAliasOutput_1 = require("../shapes/GetAliasOutput");
var ServiceException_1 = require("../shapes/ServiceException");
var ResourceNotFoundException_1 = require("../shapes/ResourceNotFoundException");
var InvalidParameterValueException_1 = require("../shapes/InvalidParameterValueException");
var TooManyRequestsException_1 = require("../shapes/TooManyRequestsException");
var ServiceMetadata_1 = require("../ServiceMetadata");
exports.GetAlias = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: "GetAlias",
    http: {
        method: "GET",
        requestUri: "/2015-03-31/functions/{FunctionName}/aliases/{Name}"
    },
    input: {
        shape: GetAliasInput_1.GetAliasInput
    },
    output: {
        shape: GetAliasOutput_1.GetAliasOutput
    },
    errors: [
        {
            shape: ServiceException_1.ServiceException
        },
        {
            shape: ResourceNotFoundException_1.ResourceNotFoundException
        },
        {
            shape: InvalidParameterValueException_1.InvalidParameterValueException
        },
        {
            shape: TooManyRequestsException_1.TooManyRequestsException
        }
    ]
};
