// @aws-sdk/client-lambda-browser/model/operations/GetAccountSettings.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var GetAccountSettingsInput_1 = require("../shapes/GetAccountSettingsInput");
var GetAccountSettingsOutput_1 = require("../shapes/GetAccountSettingsOutput");
var TooManyRequestsException_1 = require("../shapes/TooManyRequestsException");
var ServiceException_1 = require("../shapes/ServiceException");
var ServiceMetadata_1 = require("../ServiceMetadata");
exports.GetAccountSettings = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: "GetAccountSettings",
    http: {
        method: "GET",
        requestUri: "/2016-08-19/account-settings/"
    },
    input: {
        shape: GetAccountSettingsInput_1.GetAccountSettingsInput
    },
    output: {
        shape: GetAccountSettingsOutput_1.GetAccountSettingsOutput
    },
    errors: [
        {
            shape: TooManyRequestsException_1.TooManyRequestsException
        },
        {
            shape: ServiceException_1.ServiceException
        }
    ]
};
