// @aws-sdk/client-lambda-browser/model/operations/GetEventSourceMapping.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var GetEventSourceMappingInput_1 = require("../shapes/GetEventSourceMappingInput");
var GetEventSourceMappingOutput_1 = require("../shapes/GetEventSourceMappingOutput");
var ServiceException_1 = require("../shapes/ServiceException");
var ResourceNotFoundException_1 = require("../shapes/ResourceNotFoundException");
var InvalidParameterValueException_1 = require("../shapes/InvalidParameterValueException");
var TooManyRequestsException_1 = require("../shapes/TooManyRequestsException");
var ServiceMetadata_1 = require("../ServiceMetadata");
exports.GetEventSourceMapping = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: "GetEventSourceMapping",
    http: {
        method: "GET",
        requestUri: "/2015-03-31/event-source-mappings/{UUID}"
    },
    input: {
        shape: GetEventSourceMappingInput_1.GetEventSourceMappingInput
    },
    output: {
        shape: GetEventSourceMappingOutput_1.GetEventSourceMappingOutput
    },
    errors: [
        {
            shape: ServiceException_1.ServiceException
        },
        {
            shape: ResourceNotFoundException_1.ResourceNotFoundException
        },
        {
            shape: InvalidParameterValueException_1.InvalidParameterValueException
        },
        {
            shape: TooManyRequestsException_1.TooManyRequestsException
        }
    ]
};
