// @aws-sdk/client-lambda-browser/model/operations/InvokeAsync.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var InvokeAsyncInput_1 = require("../shapes/InvokeAsyncInput");
var InvokeAsyncOutput_1 = require("../shapes/InvokeAsyncOutput");
var ServiceException_1 = require("../shapes/ServiceException");
var ResourceNotFoundException_1 = require("../shapes/ResourceNotFoundException");
var InvalidRequestContentException_1 = require("../shapes/InvalidRequestContentException");
var InvalidRuntimeException_1 = require("../shapes/InvalidRuntimeException");
var ServiceMetadata_1 = require("../ServiceMetadata");
exports.InvokeAsync = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: "InvokeAsync",
    http: {
        method: "POST",
        requestUri: "/2014-11-13/functions/{FunctionName}/invoke-async/"
    },
    input: {
        shape: InvokeAsyncInput_1.InvokeAsyncInput
    },
    output: {
        shape: InvokeAsyncOutput_1.InvokeAsyncOutput
    },
    errors: [
        {
            shape: ServiceException_1.ServiceException
        },
        {
            shape: ResourceNotFoundException_1.ResourceNotFoundException
        },
        {
            shape: InvalidRequestContentException_1.InvalidRequestContentException
        },
        {
            shape: InvalidRuntimeException_1.InvalidRuntimeException
        }
    ]
};
