// @aws-sdk/client-lambda-browser/model/operations/ListLayerVersions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ListLayerVersionsInput_1 = require("../shapes/ListLayerVersionsInput");
var ListLayerVersionsOutput_1 = require("../shapes/ListLayerVersionsOutput");
var ServiceException_1 = require("../shapes/ServiceException");
var InvalidParameterValueException_1 = require("../shapes/InvalidParameterValueException");
var ResourceNotFoundException_1 = require("../shapes/ResourceNotFoundException");
var TooManyRequestsException_1 = require("../shapes/TooManyRequestsException");
var ServiceMetadata_1 = require("../ServiceMetadata");
exports.ListLayerVersions = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: "ListLayerVersions",
    http: {
        method: "GET",
        requestUri: "/2018-10-31/layers/{LayerName}/versions"
    },
    input: {
        shape: ListLayerVersionsInput_1.ListLayerVersionsInput
    },
    output: {
        shape: ListLayerVersionsOutput_1.ListLayerVersionsOutput
    },
    errors: [
        {
            shape: ServiceException_1.ServiceException
        },
        {
            shape: InvalidParameterValueException_1.InvalidParameterValueException
        },
        {
            shape: ResourceNotFoundException_1.ResourceNotFoundException
        },
        {
            shape: TooManyRequestsException_1.TooManyRequestsException
        }
    ]
};
