// @aws-sdk/client-lambda-browser/model/operations/ListFunctions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ListFunctionsInput_1 = require("../shapes/ListFunctionsInput");
var ListFunctionsOutput_1 = require("../shapes/ListFunctionsOutput");
var ServiceException_1 = require("../shapes/ServiceException");
var TooManyRequestsException_1 = require("../shapes/TooManyRequestsException");
var InvalidParameterValueException_1 = require("../shapes/InvalidParameterValueException");
var ServiceMetadata_1 = require("../ServiceMetadata");
exports.ListFunctions = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: "ListFunctions",
    http: {
        method: "GET",
        requestUri: "/2015-03-31/functions/"
    },
    input: {
        shape: ListFunctionsInput_1.ListFunctionsInput
    },
    output: {
        shape: ListFunctionsOutput_1.ListFunctionsOutput
    },
    errors: [
        {
            shape: ServiceException_1.ServiceException
        },
        {
            shape: TooManyRequestsException_1.TooManyRequestsException
        },
        {
            shape: InvalidParameterValueException_1.InvalidParameterValueException
        }
    ]
};
