// @aws-sdk/client-lambda-browser/model/operations/ListAliases.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ListAliasesInput_1 = require("../shapes/ListAliasesInput");
var ListAliasesOutput_1 = require("../shapes/ListAliasesOutput");
var ServiceException_1 = require("../shapes/ServiceException");
var ResourceNotFoundException_1 = require("../shapes/ResourceNotFoundException");
var InvalidParameterValueException_1 = require("../shapes/InvalidParameterValueException");
var TooManyRequestsException_1 = require("../shapes/TooManyRequestsException");
var ServiceMetadata_1 = require("../ServiceMetadata");
exports.ListAliases = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: "ListAliases",
    http: {
        method: "GET",
        requestUri: "/2015-03-31/functions/{FunctionName}/aliases"
    },
    input: {
        shape: ListAliasesInput_1.ListAliasesInput
    },
    output: {
        shape: ListAliasesOutput_1.ListAliasesOutput
    },
    errors: [
        {
            shape: ServiceException_1.ServiceException
        },
        {
            shape: ResourceNotFoundException_1.ResourceNotFoundException
        },
        {
            shape: InvalidParameterValueException_1.InvalidParameterValueException
        },
        {
            shape: TooManyRequestsException_1.TooManyRequestsException
        }
    ]
};
