// @aws-sdk/client-lambda-browser/model/operations/ListVersionsByFunction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ListVersionsByFunctionInput_1 = require("../shapes/ListVersionsByFunctionInput");
var ListVersionsByFunctionOutput_1 = require("../shapes/ListVersionsByFunctionOutput");
var ServiceException_1 = require("../shapes/ServiceException");
var ResourceNotFoundException_1 = require("../shapes/ResourceNotFoundException");
var InvalidParameterValueException_1 = require("../shapes/InvalidParameterValueException");
var TooManyRequestsException_1 = require("../shapes/TooManyRequestsException");
var ServiceMetadata_1 = require("../ServiceMetadata");
exports.ListVersionsByFunction = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: "ListVersionsByFunction",
    http: {
        method: "GET",
        requestUri: "/2015-03-31/functions/{FunctionName}/versions"
    },
    input: {
        shape: ListVersionsByFunctionInput_1.ListVersionsByFunctionInput
    },
    output: {
        shape: ListVersionsByFunctionOutput_1.ListVersionsByFunctionOutput
    },
    errors: [
        {
            shape: ServiceException_1.ServiceException
        },
        {
            shape: ResourceNotFoundException_1.ResourceNotFoundException
        },
        {
            shape: InvalidParameterValueException_1.InvalidParameterValueException
        },
        {
            shape: TooManyRequestsException_1.TooManyRequestsException
        }
    ]
};
