// @aws-sdk/client-lambda-browser/model/operations/ListEventSourceMappings.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ListEventSourceMappingsInput_1 = require("../shapes/ListEventSourceMappingsInput");
var ListEventSourceMappingsOutput_1 = require("../shapes/ListEventSourceMappingsOutput");
var ServiceException_1 = require("../shapes/ServiceException");
var ResourceNotFoundException_1 = require("../shapes/ResourceNotFoundException");
var InvalidParameterValueException_1 = require("../shapes/InvalidParameterValueException");
var TooManyRequestsException_1 = require("../shapes/TooManyRequestsException");
var ServiceMetadata_1 = require("../ServiceMetadata");
exports.ListEventSourceMappings = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: "ListEventSourceMappings",
    http: {
        method: "GET",
        requestUri: "/2015-03-31/event-source-mappings/"
    },
    input: {
        shape: ListEventSourceMappingsInput_1.ListEventSourceMappingsInput
    },
    output: {
        shape: ListEventSourceMappingsOutput_1.ListEventSourceMappingsOutput
    },
    errors: [
        {
            shape: ServiceException_1.ServiceException
        },
        {
            shape: ResourceNotFoundException_1.ResourceNotFoundException
        },
        {
            shape: InvalidParameterValueException_1.InvalidParameterValueException
        },
        {
            shape: TooManyRequestsException_1.TooManyRequestsException
        }
    ]
};
