// @aws-sdk/client-lambda-browser/model/operations/PublishLayerVersion.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var PublishLayerVersionInput_1 = require("../shapes/PublishLayerVersionInput");
var PublishLayerVersionOutput_1 = require("../shapes/PublishLayerVersionOutput");
var ServiceException_1 = require("../shapes/ServiceException");
var ResourceNotFoundException_1 = require("../shapes/ResourceNotFoundException");
var TooManyRequestsException_1 = require("../shapes/TooManyRequestsException");
var InvalidParameterValueException_1 = require("../shapes/InvalidParameterValueException");
var CodeStorageExceededException_1 = require("../shapes/CodeStorageExceededException");
var ServiceMetadata_1 = require("../ServiceMetadata");
exports.PublishLayerVersion = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: "PublishLayerVersion",
    http: {
        method: "POST",
        requestUri: "/2018-10-31/layers/{LayerName}/versions"
    },
    input: {
        shape: PublishLayerVersionInput_1.PublishLayerVersionInput
    },
    output: {
        shape: PublishLayerVersionOutput_1.PublishLayerVersionOutput
    },
    errors: [
        {
            shape: ServiceException_1.ServiceException
        },
        {
            shape: ResourceNotFoundException_1.ResourceNotFoundException
        },
        {
            shape: TooManyRequestsException_1.TooManyRequestsException
        },
        {
            shape: InvalidParameterValueException_1.InvalidParameterValueException
        },
        {
            shape: CodeStorageExceededException_1.CodeStorageExceededException
        }
    ]
};
