// @aws-sdk/client-lambda-browser/model/operations/PublishVersion.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var PublishVersionInput_1 = require("../shapes/PublishVersionInput");
var PublishVersionOutput_1 = require("../shapes/PublishVersionOutput");
var ServiceException_1 = require("../shapes/ServiceException");
var ResourceNotFoundException_1 = require("../shapes/ResourceNotFoundException");
var InvalidParameterValueException_1 = require("../shapes/InvalidParameterValueException");
var TooManyRequestsException_1 = require("../shapes/TooManyRequestsException");
var CodeStorageExceededException_1 = require("../shapes/CodeStorageExceededException");
var PreconditionFailedException_1 = require("../shapes/PreconditionFailedException");
var ServiceMetadata_1 = require("../ServiceMetadata");
exports.PublishVersion = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: "PublishVersion",
    http: {
        method: "POST",
        requestUri: "/2015-03-31/functions/{FunctionName}/versions"
    },
    input: {
        shape: PublishVersionInput_1.PublishVersionInput
    },
    output: {
        shape: PublishVersionOutput_1.PublishVersionOutput
    },
    errors: [
        {
            shape: ServiceException_1.ServiceException
        },
        {
            shape: ResourceNotFoundException_1.ResourceNotFoundException
        },
        {
            shape: InvalidParameterValueException_1.InvalidParameterValueException
        },
        {
            shape: TooManyRequestsException_1.TooManyRequestsException
        },
        {
            shape: CodeStorageExceededException_1.CodeStorageExceededException
        },
        {
            shape: PreconditionFailedException_1.PreconditionFailedException
        }
    ]
};
