// @aws-sdk/client-lambda-browser/model/operations/RemovePermission.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var RemovePermissionInput_1 = require("../shapes/RemovePermissionInput");
var RemovePermissionOutput_1 = require("../shapes/RemovePermissionOutput");
var ServiceException_1 = require("../shapes/ServiceException");
var ResourceNotFoundException_1 = require("../shapes/ResourceNotFoundException");
var InvalidParameterValueException_1 = require("../shapes/InvalidParameterValueException");
var TooManyRequestsException_1 = require("../shapes/TooManyRequestsException");
var PreconditionFailedException_1 = require("../shapes/PreconditionFailedException");
var ServiceMetadata_1 = require("../ServiceMetadata");
exports.RemovePermission = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: "RemovePermission",
    http: {
        method: "DELETE",
        requestUri: "/2015-03-31/functions/{FunctionName}/policy/{StatementId}"
    },
    input: {
        shape: RemovePermissionInput_1.RemovePermissionInput
    },
    output: {
        shape: RemovePermissionOutput_1.RemovePermissionOutput
    },
    errors: [
        {
            shape: ServiceException_1.ServiceException
        },
        {
            shape: ResourceNotFoundException_1.ResourceNotFoundException
        },
        {
            shape: InvalidParameterValueException_1.InvalidParameterValueException
        },
        {
            shape: TooManyRequestsException_1.TooManyRequestsException
        },
        {
            shape: PreconditionFailedException_1.PreconditionFailedException
        }
    ]
};
