// @aws-sdk/client-lambda-browser/model/operations/TagResource.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var TagResourceInput_1 = require("../shapes/TagResourceInput");
var TagResourceOutput_1 = require("../shapes/TagResourceOutput");
var ServiceException_1 = require("../shapes/ServiceException");
var ResourceNotFoundException_1 = require("../shapes/ResourceNotFoundException");
var InvalidParameterValueException_1 = require("../shapes/InvalidParameterValueException");
var TooManyRequestsException_1 = require("../shapes/TooManyRequestsException");
var ServiceMetadata_1 = require("../ServiceMetadata");
exports.TagResource = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: "TagResource",
    http: {
        method: "POST",
        requestUri: "/2017-03-31/tags/{ARN}"
    },
    input: {
        shape: TagResourceInput_1.TagResourceInput
    },
    output: {
        shape: TagResourceOutput_1.TagResourceOutput
    },
    errors: [
        {
            shape: ServiceException_1.ServiceException
        },
        {
            shape: ResourceNotFoundException_1.ResourceNotFoundException
        },
        {
            shape: InvalidParameterValueException_1.InvalidParameterValueException
        },
        {
            shape: TooManyRequestsException_1.TooManyRequestsException
        }
    ]
};
