// @aws-sdk/client-lambda-browser/model/operations/UpdateAlias.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var UpdateAliasInput_1 = require("../shapes/UpdateAliasInput");
var UpdateAliasOutput_1 = require("../shapes/UpdateAliasOutput");
var ServiceException_1 = require("../shapes/ServiceException");
var ResourceNotFoundException_1 = require("../shapes/ResourceNotFoundException");
var InvalidParameterValueException_1 = require("../shapes/InvalidParameterValueException");
var TooManyRequestsException_1 = require("../shapes/TooManyRequestsException");
var PreconditionFailedException_1 = require("../shapes/PreconditionFailedException");
var ServiceMetadata_1 = require("../ServiceMetadata");
exports.UpdateAlias = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: "UpdateAlias",
    http: {
        method: "PUT",
        requestUri: "/2015-03-31/functions/{FunctionName}/aliases/{Name}"
    },
    input: {
        shape: UpdateAliasInput_1.UpdateAliasInput
    },
    output: {
        shape: UpdateAliasOutput_1.UpdateAliasOutput
    },
    errors: [
        {
            shape: ServiceException_1.ServiceException
        },
        {
            shape: ResourceNotFoundException_1.ResourceNotFoundException
        },
        {
            shape: InvalidParameterValueException_1.InvalidParameterValueException
        },
        {
            shape: TooManyRequestsException_1.TooManyRequestsException
        },
        {
            shape: PreconditionFailedException_1.PreconditionFailedException
        }
    ]
};
