// @aws-sdk/client-lambda-browser/commands/DeleteFunctionConcurrencyCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var __aws_sdk_middleware_stack = require("@aws-sdk/middleware-stack");
var DeleteFunctionConcurrency_1 = require("../model/operations/DeleteFunctionConcurrency");
var DeleteFunctionConcurrencyCommand = /** @class */ (function () {
    function DeleteFunctionConcurrencyCommand(input) {
        this.input = input;
        this.model = DeleteFunctionConcurrency_1.DeleteFunctionConcurrency;
        this.middlewareStack = new __aws_sdk_middleware_stack.MiddlewareStack();
    }
    DeleteFunctionConcurrencyCommand.prototype.resolveMiddleware = function (clientStack, configuration) {
        var handler = configuration.handler;
        var stack = clientStack.concat(this.middlewareStack);
        var handlerExecutionContext = {
            logger: {},
            model: this.model
        };
        return stack.resolve(handler(handlerExecutionContext), handlerExecutionContext);
    };
    return DeleteFunctionConcurrencyCommand;
}());
exports.DeleteFunctionConcurrencyCommand = DeleteFunctionConcurrencyCommand;
