// @aws-sdk/client-lambda-browser/commands/DeleteEventSourceMappingCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var __aws_sdk_middleware_stack = require("@aws-sdk/middleware-stack");
var DeleteEventSourceMapping_1 = require("../model/operations/DeleteEventSourceMapping");
var DeleteEventSourceMappingCommand = /** @class */ (function () {
    function DeleteEventSourceMappingCommand(input) {
        this.input = input;
        this.model = DeleteEventSourceMapping_1.DeleteEventSourceMapping;
        this.middlewareStack = new __aws_sdk_middleware_stack.MiddlewareStack();
    }
    DeleteEventSourceMappingCommand.prototype.resolveMiddleware = function (clientStack, configuration) {
        var handler = configuration.handler;
        var stack = clientStack.concat(this.middlewareStack);
        var handlerExecutionContext = {
            logger: {},
            model: this.model
        };
        return stack.resolve(handler(handlerExecutionContext), handlerExecutionContext);
    };
    return DeleteEventSourceMappingCommand;
}());
exports.DeleteEventSourceMappingCommand = DeleteEventSourceMappingCommand;
