// @aws-sdk/client-lambda-browser/commands/GetEventSourceMappingCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var __aws_sdk_middleware_stack = require("@aws-sdk/middleware-stack");
var GetEventSourceMapping_1 = require("../model/operations/GetEventSourceMapping");
var GetEventSourceMappingCommand = /** @class */ (function () {
    function GetEventSourceMappingCommand(input) {
        this.input = input;
        this.model = GetEventSourceMapping_1.GetEventSourceMapping;
        this.middlewareStack = new __aws_sdk_middleware_stack.MiddlewareStack();
    }
    GetEventSourceMappingCommand.prototype.resolveMiddleware = function (clientStack, configuration) {
        var handler = configuration.handler;
        var stack = clientStack.concat(this.middlewareStack);
        var handlerExecutionContext = {
            logger: {},
            model: this.model
        };
        return stack.resolve(handler(handlerExecutionContext), handlerExecutionContext);
    };
    return GetEventSourceMappingCommand;
}());
exports.GetEventSourceMappingCommand = GetEventSourceMappingCommand;
