// @aws-sdk/client-lambda-browser/commands/GetFunctionConfigurationCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var __aws_sdk_middleware_stack = require("@aws-sdk/middleware-stack");
var GetFunctionConfiguration_1 = require("../model/operations/GetFunctionConfiguration");
var GetFunctionConfigurationCommand = /** @class */ (function () {
    function GetFunctionConfigurationCommand(input) {
        this.input = input;
        this.model = GetFunctionConfiguration_1.GetFunctionConfiguration;
        this.middlewareStack = new __aws_sdk_middleware_stack.MiddlewareStack();
    }
    GetFunctionConfigurationCommand.prototype.resolveMiddleware = function (clientStack, configuration) {
        var handler = configuration.handler;
        var stack = clientStack.concat(this.middlewareStack);
        var handlerExecutionContext = {
            logger: {},
            model: this.model
        };
        return stack.resolve(handler(handlerExecutionContext), handlerExecutionContext);
    };
    return GetFunctionConfigurationCommand;
}());
exports.GetFunctionConfigurationCommand = GetFunctionConfigurationCommand;
