// @aws-sdk/client-lambda-browser/commands/ListVersionsByFunctionCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var __aws_sdk_middleware_stack = require("@aws-sdk/middleware-stack");
var ListVersionsByFunction_1 = require("../model/operations/ListVersionsByFunction");
var ListVersionsByFunctionCommand = /** @class */ (function () {
    function ListVersionsByFunctionCommand(input) {
        this.input = input;
        this.model = ListVersionsByFunction_1.ListVersionsByFunction;
        this.middlewareStack = new __aws_sdk_middleware_stack.MiddlewareStack();
    }
    ListVersionsByFunctionCommand.prototype.resolveMiddleware = function (clientStack, configuration) {
        var handler = configuration.handler;
        var stack = clientStack.concat(this.middlewareStack);
        var handlerExecutionContext = {
            logger: {},
            model: this.model
        };
        return stack.resolve(handler(handlerExecutionContext), handlerExecutionContext);
    };
    return ListVersionsByFunctionCommand;
}());
exports.ListVersionsByFunctionCommand = ListVersionsByFunctionCommand;
