// @aws-sdk/client-lambda-browser/commands/PublishLayerVersionCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var __aws_sdk_middleware_stack = require("@aws-sdk/middleware-stack");
var PublishLayerVersion_1 = require("../model/operations/PublishLayerVersion");
var PublishLayerVersionCommand = /** @class */ (function () {
    function PublishLayerVersionCommand(input) {
        this.input = input;
        this.model = PublishLayerVersion_1.PublishLayerVersion;
        this.middlewareStack = new __aws_sdk_middleware_stack.MiddlewareStack();
    }
    PublishLayerVersionCommand.prototype.resolveMiddleware = function (clientStack, configuration) {
        var handler = configuration.handler;
        var stack = clientStack.concat(this.middlewareStack);
        var handlerExecutionContext = {
            logger: {},
            model: this.model
        };
        return stack.resolve(handler(handlerExecutionContext), handlerExecutionContext);
    };
    return PublishLayerVersionCommand;
}());
exports.PublishLayerVersionCommand = PublishLayerVersionCommand;
