// @aws-sdk/client-lambda-browser/commands/UpdateFunctionConfigurationCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var __aws_sdk_middleware_stack = require("@aws-sdk/middleware-stack");
var UpdateFunctionConfiguration_1 = require("../model/operations/UpdateFunctionConfiguration");
var UpdateFunctionConfigurationCommand = /** @class */ (function () {
    function UpdateFunctionConfigurationCommand(input) {
        this.input = input;
        this.model = UpdateFunctionConfiguration_1.UpdateFunctionConfiguration;
        this.middlewareStack = new __aws_sdk_middleware_stack.MiddlewareStack();
    }
    UpdateFunctionConfigurationCommand.prototype.resolveMiddleware = function (clientStack, configuration) {
        var handler = configuration.handler;
        var stack = clientStack.concat(this.middlewareStack);
        var handlerExecutionContext = {
            logger: {},
            model: this.model
        };
        return stack.resolve(handler(handlerExecutionContext), handlerExecutionContext);
    };
    return UpdateFunctionConfigurationCommand;
}());
exports.UpdateFunctionConfigurationCommand = UpdateFunctionConfigurationCommand;
