// @aws-sdk/client-lambda-browser/commands/UpdateFunctionCodeCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var __aws_sdk_middleware_stack = require("@aws-sdk/middleware-stack");
var UpdateFunctionCode_1 = require("../model/operations/UpdateFunctionCode");
var UpdateFunctionCodeCommand = /** @class */ (function () {
    function UpdateFunctionCodeCommand(input) {
        this.input = input;
        this.model = UpdateFunctionCode_1.UpdateFunctionCode;
        this.middlewareStack = new __aws_sdk_middleware_stack.MiddlewareStack();
    }
    UpdateFunctionCodeCommand.prototype.resolveMiddleware = function (clientStack, configuration) {
        var handler = configuration.handler;
        var stack = clientStack.concat(this.middlewareStack);
        var handlerExecutionContext = {
            logger: {},
            model: this.model
        };
        return stack.resolve(handler(handlerExecutionContext), handlerExecutionContext);
    };
    return UpdateFunctionCodeCommand;
}());
exports.UpdateFunctionCodeCommand = UpdateFunctionCodeCommand;
